base/reflection.jl

```diff
@@ -179,6 +194,10 @@ function methods(f::ANY,t::ANY)
     Any[m[3] for m in _methods(f,t,-1)]
 end
 function _methods(f::ANY,t::ANY,lim)
+    #special case Expr to allow methods(@time,Tuple{Any})
+    if _is_empty_macrocall(f)
+        f = _get_macro(current_module(),f.args[1])
```

Comment:
I do not quite follow you. The meaning of the function is to get function object of the macro from the `getfield` expression. Do you have an alternative way of obtaining the result?